apps/mobile/src/features/TxHistory/components/TxHistoryList/TxHistoryList.tsx

```diff
@@ -1,62 +1,196 @@
   onEndReached: (info: { distanceFromEnd: number }) => void
   isLoading?: boolean
+  isLoadingNext?: boolean
+  isLoadingPrevious?: boolean
   refreshing?: boolean
   onRefresh?: () => void
 }
 
-export function TxHistoryList({ transactions, onEndReached, isLoading, refreshing, onRefresh }: TxHistoryList) {
-  const theme = useTheme()
-
-  const groupedList: GroupedTxsWithTitle<TransactionItem>[] = useMemo(() => {
-    return groupTxsByDate(transactions || [])
-  }, [transactions])
-
-  const hasTransactions = transactions && transactions.length > 0
-  const isInitialLoading = isLoading && !hasTransactions && !refreshing
+const TAB_BAR_HEIGHT = 34
 
-  // ListEmptyComponent for initial loading state
-  const renderEmptyComponent = useMemo(() => {
-    if (isInitialLoading) {
+const renderItem = ({
+  item,
+  target,
+}: {
+  item: HistoryTransactionItems | HistoryTransactionItems[]
+  target?: string
+}) => {
+  if (Array.isArray(item)) {
+    // Render grouped transactions - filter to only TransactionItems for TxGroupedCard
+    const transactionItems = item.filter((tx): tx is TransactionItem => tx.type === 'TRANSACTION')
+    if (transactionItems.length > 0) {
       return (
-        <View
-          flex={1}
-          alignItems="flex-start"
-          justifyContent="flex-start"
-          paddingTop="$4"
-          testID="tx-history-initial-loader"
-        >
-          <TransactionSkeleton count={6} sectionTitles={['Recent transactions']} />
+        <View marginTop="$4">
+          <TxGroupedCard transactions={transactionItems} />
         </View>
       )
     }
     return null
-  }, [isInitialLoading])
+  }
 
-  // ListFooterComponent for pagination loading (bottom loading)
-  const renderFooterComponent = useMemo(() => {
-    if (isLoading && hasTransactions) {
-      return (
-        <View testID="tx-history-pagination-loader" marginTop="$4">
-          <TransactionSkeletonItem />
-        </View>
-      )
-    }
-    return null
-  }, [isLoading, hasTransactions])
+  if (isDateLabel(item)) {
+    const dateTitle = formatWithSchema(item.timestamp, 'MMM d, yyyy')
+    const isSticky = target === 'StickyHeader'
+
+    return (
+      <View
+        marginTop={isSticky ? '$0' : '$2'}
+        backgroundColor={'$background'}
+        paddingTop={isSticky ? '$2' : '$2'}
+        paddingBottom={isSticky ? '$2' : '0'}
+        paddingHorizontal={isSticky ? '$4' : '0'}
+        transform={[{ translateY: isSticky ? TAB_BAR_HEIGHT : 0 }]}
+      >
+        <Text fontWeight={500} color="$colorSecondary">
+          {dateTitle}
+        </Text>
+      </View>
+    )
+  }
+
+  if (item.type === 'TRANSACTION') {
+    return (
+      <View marginTop="$4">
+        <TxInfo tx={item.transaction} />
+      </View>
+    )
+  }
+
+  // Fallback for other item types
+  return (
+    <View backgroundColor="lightgray" height={80} marginBottom={16} marginHorizontal={16}>
+      <Text>Unknown item type</Text>
+    </View>
+  )
+}
```

Comment:
nit: How can this fallback happen? It does look different than the other cards:
<img width="379" height="316" alt="Image" src="https://github.com/user-attachments/assets/2dd4a94b-6475-488e-84ae-d0233b82d333" />